composer.json

```diff
@@ -12,20 +11,24 @@
         }
     ],
     "require": {
-        "php": ">=5.5.0"
+        "php": "^5.5 || ^7.0"
```

Comment:
As far as I can see the problem is exactly where @GrahamCampbell pointed: whenever a new PHP major is out AND there is a break composer will simply install a version which is known not to work with PHP. Even if you add a patch which specifically disallows the next major, you are already late since composer will simply install an older version in the same series, which will not work, but composer does not know that. You cannot redo all past tags with version compatibility updated. So the solution is to have an upper limit for the versions you support, which should be the known compatible versions. I understand that it might be uncomfortable, but this way there are no issues when it comes to compatibility with a later major version which might break BC. Very rarely is not something that I trust. 😉 But as @GrahamCampbell we are already late with this change, since adding the upper limit will not prevent the issue from happening, so doesn't really make sense to add this change in the current release series.